src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/OnAutoInsertHandler.cs

```diff
@@ -109,21 +109,48 @@ public async Task<DocumentOnAutoInsertResponseItem> HandleRequestAsync(DocumentO
                 return null;
             }
 
-            var remappedEdit = await _documentMappingProvider.RemapFormattedTextEditsAsync(
+            var containsSnippet = response.TextEditFormat == InsertTextFormat.Snippet;
+
+            var edit = response.TextEdit;
+            if (containsSnippet && projectionResult.LanguageKind == RazorLanguageKind.CSharp)
+            {
+                // Formatting doesn't work with syntax errors caused by the cursor marker ($0).
+                // So, let's avoid the error by wrapping the cursor marker in a comment.
```

Comment:
I know this is ugly but IMO this is less ugly compared to special-casing this knowledge in the formatter and handling this on the server.
Having said that, I know this is still not an acceptable long term solution. I'm hoping to address this while working on https://github.com/dotnet/aspnetcore/issues/27144.